lib/metadata/bucketfile/backend.js

```diff
@@ -29,152 +19,42 @@ class BucketFileInterface {
     constructor() {
         this.logger = logger;
         if (cluster.isMaster) {
-            this.startServer();
-        } else {
-            this.refcnt = 0;
-            this.waitreco = 0;
-            this.realReConnect();
-            this.notifier = new events.EventEmitter();
+            this.mdServer = new MetadataServer(
```

Comment:
I think you should be able to start the metadata server in `init.js`; this would make metadata, data, and S3 more separated and it's easier to deploy more S3 to use the same existing metadata and data backends.